browser/module.go

```diff
@@ -71,9 +71,17 @@ func (*RootModule) NewModuleInstance(vu k6modules.VU) k6modules.Instance {
 		k6common.Throw(vu.Runtime(), errors.New(msg))
 	}
 
+	// using our custom VU so that we can be ready for the future
+	// changes to the k6-core VU code. And we can have a fine-grained
+	// control over it, now and in the future.
+	//
+	// this puts the VU object in the context so that it can be
+	// used by inner objects such as k6ext.Promise.
+	vu = moduleVU{vu}
```

Comment:
@ankur22 This is my rationale for using our own `moduleVU` instead of a plain VU.

Reply:
It's stepping into the "over engineering" realm. Have we needed this before?

I'd be happy with it if this is all we ever needed to interact with and when we interacted with `vu` it would abstract away any of the `k6modules.VU` internals.

Reply:
Fair enough. See: https://github.com/grafana/xk6-browser/pull/704/commits/5822656b1d4f06f92eae38eb8334122b54d39afe